core/domain/exp_domain.py

```diff
@@ -2514,12 +2515,1064 @@ def _convert_states_v50_dict_to_v51_dict(
+            - If refresher_exploration_id is not None for lesson, marks it as
+            None
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for state_name, state_dict in states_dict.items():
+            answer_groups = state_dict['interaction']['answer_groups']
+            for answer_group in answer_groups:
+                # labelled_as_correct should not be True if dest is try again.
+                if (
+                    answer_group['outcome']['dest'] == state_name and
+                    answer_group['outcome']['labelled_as_correct']
+                ):
+                    answer_group['outcome']['labelled_as_correct'] = False
+
+                # refresher_exploration_id be None for all lessons.
+                answer_group['outcome']['refresher_exploration_id'] = None
+
+            state_dict['interaction']['answer_groups'] = answer_groups
+
+            # refresher_exploration_id be None for all lessons(def outcome).
+            default_outcome = state_dict['interaction']['default_outcome']
+            if default_outcome is not None:
+                state_dict['interaction']['default_outcome'][
+                    'refresher_exploration_id'] = None
+
+        return states_dict
+
+    @classmethod
+    def _choices_should_be_unique_and_non_empty(
+        cls, choices: List[str],
+        answer_groups: List[state_domain.AnswerGroup]
+    ) -> Tuple(Any):
+        """Handles choices present in the ItemSelectionInput or
+        in MultipleChoiceInput interactions, implements the following -
+            - If only one choice is empty then simply removes it
+            - If multiple choices are empty replace them with `Choice 1` ,
+            `Choice 2` etc
+            - If choices are duplicate, removes the later choice
+
+        Args:
+            choices: List[str]. A list of choices.
+
+        Returns:
+            choices_to_save: List[str]. The list of valid choices.
+        """
+        # Misbehaving when after removing invalid choices and the choices count becomes below 4.
+        empty_choices = []
+        seen_choices = []
+        choices_to_remove = []
+        invalid_choices_index = []
+        for choice in choices:
+            if choice['html'].strip() in ('<p></p>', ''):
+                empty_choices.append(choice)
+
+        # Only one choice is empty.
+        if len(empty_choices) == 1:
+            invalid_choices_index.append(choices.index(empty_choices[0]))
+            choices.remove(empty_choices[0])
+        # Multiple choices are empty.
+        else:
+            for idx, empty_choice in enumerate(empty_choices):
+                empty_choice['html'] = (
+                    '<p>' + 'Choice ' + str(idx+1) + '</p>'
+                )
```

Comment:
And are the empty choices in the original list of choices actually fixed by this code? You never modify the `choices` argument, right?